src/vignette/util/thumbnail.clj

```diff
@@ -79,6 +79,7 @@
 
 (defn get-or-generate-thumbnail
   [system thumb-map]
+  (assert-original-mime-type (get thumb-map :original) thumb-map)
```

Comment:
` (u/assert-original-mime-type "file" {}) ` does not throw an exception. I should add a test for that.